docs/key_concepts.rst

```diff
@@ -0,0 +1,83 @@
+Key Concepts
+============
+
+This section introduces nameko's central concepts.
+
+Anatomy of a Service
+--------------------
+
+A nameko service is just a Python class. The class encapsulates the application logic in its methods and declares any :ref:`dependencies <dependencies>` as attributes.
+
+Methods are exposed to the outside world with :ref:`entrypoint <entrypoints>` decorators.
+
+.. literalinclude:: examples/example.py
+
+.. _entrypoints:
+
+Entrypoints
+^^^^^^^^^^^
+
+Entrypoints are gateways into the service methods they decorate. While a service is running, an entrypoint may "fire" and the decorated method would be executed by a service :ref:`worker <workers>`.
+
+.. _dependencies:
+
+Dependencies
+^^^^^^^^^^^^
+
+Most services depend on something other than themselves. Nameko encourages these things to be implemented as dependencies.
+
+A dependency is an opportunity to hide code that isn't part of the core service logic. The dependency's interface to the service should be as simple as possible.
+
+Declaring dependencies in your service is a good idea for :ref:`lots of reasons <benefits_of_dependency_injection>`, and you should think of them as the gateway between service code and everything else. That includes other services, external APIs, and even databases.
```

Comment:
It maybe appropriate to mention at some point in this document something like "Nameko includes a bunch of built-in entrypoints and dependencies" (and then a link to the built-in extensions) "or you can make your own" (and the the link to writing dependencies). 

This is because I found a bit confusing that the section describing the built-in stuff is calls it "extensions", when that term isn't mentioned anywhere in this terminology doc. Including the links from here would help make the connection IMO.